include/glaze/ext/jsonrpc.hpp

```diff
@@ -0,0 +1,548 @@
+#include <glaze/glaze.hpp>
+#include <glaze/tuplet/tuple.hpp>
+#include <glaze/util/expected.hpp>
+
+#include <algorithm>
+#include <deque>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <variant>
+#include <vector>
+
+#if __cpp_lib_to_underlying
+namespace util
+{
+   using std::to_underlying;
+}
+#else
+namespace util
+{
+   template <typename T>
+   [[nodiscard]] constexpr std::underlying_type_t<T> to_underlying(T value) noexcept
+   {
+      return static_cast<std::underlying_type_t<T>>(value);
+   }
+}
+#endif
+
+namespace glz::rpc
+{
+   enum struct error_e : int {
+      no_error = 0,
+      server_error_lower = -32000,
+      server_error_upper = -32099,
+      invalid_request = -32600,
+      method_not_found = -32601,
+      invalid_params = -32602,
+      internal = -32603,
+      parse_error = -32700,
+   };
+   static constexpr std::array<error_e, 8> error_e_iterable{
+      error_e::no_error,         error_e::server_error_lower, error_e::server_error_upper, error_e::invalid_request,
+      error_e::method_not_found, error_e::invalid_params,     error_e::internal,           error_e::parse_error,
+   };
+}
+
+template <>
+struct glz::meta<glz::rpc::error_e>
+{
+   // clang-format off
+   static constexpr auto value{[](auto&& enum_value) -> auto { return util::to_underlying(enum_value); }};
+};
+// clang-format on
+
+// jsonrpc
+namespace glz::rpc
+{
+   using jsonrpc_id_type = std::variant<glz::json_t::null_t, std::string, std::int64_t>;
```

Comment:
My opinion is that if a user requires it to support fractional it can be patched by that user. Or add a PR later with ifdef clause making it configurable.